docs/reference/aggregations/metrics/top-metrics-aggregation.asciidoc

```diff
@@ -0,0 +1,281 @@
+[role="xpack"]
+[testenv="basic"]
+[[search-aggregations-metrics-top-metrics]]
+=== Top Metrics Aggregation
+
+experimental[We expect to change the response format of this aggregation as we add more features., https://github.com/elastic/elasticsearch/issues/51813]
+
+The `top_metrics` aggregation selects metrics from the document with the largest or smallest "sort"
+value. For example, This gets the value of the `v` field on the document with the largest value of `s`:
+
+[source,console,id=search-aggregations-metrics-top-metrics-simple]
+----
+POST /test/_bulk?refresh
+{"index": {}}
+{"s": 1, "v": 3.1415}
+{"index": {}}
+{"s": 2, "v": 1}
+{"index": {}}
+{"s": 3, "v": 2.71828}
+POST /test/_search?filter_path=aggregations
+{
+  "aggs": {
+    "tm": {
+      "top_metrics": {
+        "metric": {"field": "v"},
+        "sort": {"s": "desc"}
+      }
+    }
+  }
+}
+----
+
+Which returns:
+
+[source,js]
+----
+{
+  "aggregations": {
+    "tm": {
+      "top": [ {"sort": [3], "metrics": {"v": 2.718280076980591 } } ]
+    }
+  }
+}
+----
+// TESTRESPONSE
+
+`top_metrics` is fairly similar to <<search-aggregations-metrics-top-hits-aggregation, `top_hits`>>
+in spirit but because it is more limited it is able to do its job using less memory and is often
+faster.
+
+==== `sort`
+
+The `sort` field in the metric request functions exactly the same as the `sort` field in the
+<<request-body-search-sort, search>> request except:
+* It can't be used on <<binary,binary>>, <<flattened,flattened>, <<ip,ip>>,
+<<keyword,keyword>>, or <<text,text>> fields.
+* It only supports a single sort value.
+
+The metrics that the aggregation returns is the first hit that would be returned by the search
+request. So,
+
+`"sort": {"s": "desc"}`:: gets metrics from the document with the highest `s`
+`"sort": {"s": "asc"}`:: gets the metrics from the document with the lowest `s`
+`"sort": {"_geo_distance": {"location": "35.7796, -78.6382"}}`::
+  gets metrics from the documents with `location` *closest* to `35.7796, -78.6382`
+`"sort": "_score"`:: gets metrics from the document with the highest score
+
+==== `metric`
+
+At this point `metric` supports only `{"field": "field_name"}` and all metrics
+are returned as double precision floating point numbers. Expect more to
+come here.
```

Comment:
This comment might be embarassing if it is still here in 10 years :)

Reply:
:heart: